### ISSUE TITLE
Regularization

### ISSUE THREAD

**Issue by Little-Spy** [Little-Spy/ML-of-Standford-University#7]
### The problem of overfitting
![image](https://user-images.githubusercontent.com/40796319/50689636-6db79580-1065-11e9-939b-4b7a2fe15677.png)
1:'undefit'  'high bias'  2:'fit well' 3:'overfitting', 'high variance'
### Addressing overfitting:
1. Reduce number of features.
   --Manually select which features to keep.
   --Model selection algorithm.
2.Regularization
   --Keep all the features, but reduce magnitude/values if parameters theta j.
   --Works well when we have a lot of features, each of which contributes a bit to predicting y.
 ![image](https://user-images.githubusercontent.com/40796319/50690021-da7f5f80-1066-11e9-8832-05dd6f8fa5f5.png)

### Regularization cost function
   --Small values for parameters，leading 'simpler' hypothesis
   --Less prone to overfitting
![image](https://user-images.githubusercontent.com/40796319/50690517-91300f80-1068-11e9-8112-d924970a4f69.png)
**Minimize J(theta)**
![image](https://user-images.githubusercontent.com/40796319/50690568-bcb2fa00-1068-11e9-8695-11686574ca9d.png)
The λ, or lambda, is the regularization parameter. It determines how much the costs of our theta parameters are inflated. We need to choose a good choice for lambda, if lambda is too large, we can even not fit the training set.
### Regularization Linear Regression
**Gradient Descent**
![image](https://user-images.githubusercontent.com/40796319/50690973-4f07cd80-106a-11e9-9657-8d58ff44c8c2.png)
![image](https://user-images.githubusercontent.com/40796319/50691165-ca697f00-106a-11e9-81b4-5c90bd780462.png)
**Normal Equation**
![image](https://user-images.githubusercontent.com/40796319/50691448-b3775c80-106b-11e9-8b49-12dfd3f481d2.png)
**Cost function of regularization Logistic Regression :** 
![image](https://user-images.githubusercontent.com/40796319/50725026-48dd2400-1132-11e9-96d5-a4a954c80558.png)
![image](https://user-images.githubusercontent.com/40796319/50725038-6a3e1000-1132-11e9-8622-649702da8280.png)
